.github/workflows/smoke-tests.yml

```diff
@@ -36,6 +36,11 @@ jobs:
         with:
           persist-credentials: false
 
+      - name: Install Python
+        uses: actions/setup-python@65d7f2d534ac1bc67fcd62888c5f4f3d2cb2b236 # v4.7.1
+        with:
+          python-version: '3.11'
+
```

Comment:
This is fixing recent break at python/node-gyp integration.
https://github.com/nodejs/node-gyp/issues?q=is%3Aissue+ModuleNotFoundError%3A+No+module+named+%27distutils%27+is%3Aclosed


```
Link step
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info it worked if it ends with ok
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info using node-gyp@9.3.1
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info using node@18.18.2 | darwin | x64
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info find Python using Python version 3.12.0 found at "/usr/local/bin/python3"
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp http GET https://nodejs.org/download/release/v18.18.2/node-v18.18.2-headers.tar.gz
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp http 200 https://nodejs.org/download/release/v18.18.2/node-v18.18.2-headers.tar.gz
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp http GET https://nodejs.org/download/release/v18.18.2/SHASUMS256.txt
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp http 200 https://nodejs.org/download/release/v18.18.2/SHASUMS256.txt
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn /usr/local/bin/python3
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args [
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '/Users/runner/work/amplify-cli/amplify-cli/node_modules/node-gyp/gyp/gyp_main.py',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   'binding.gyp',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '-f',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   'make',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '-I',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '/Users/runner/work/amplify-cli/amplify-cli/node_modules/node-pty/build/config.gypi',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '-I',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '/Users/runner/work/amplify-cli/amplify-cli/node_modules/node-gyp/addon.gypi',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '-I',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '/Users/runner/Library/Caches/node-gyp/18.18.2/include/node/common.gypi',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '-Dlibrary=shared_library',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '-Dvisibility=default',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '-Dnode_root_dir=/Users/runner/Library/Caches/node-gyp/18.18.2',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '-Dnode_gyp_dir=/Users/runner/work/amplify-cli/amplify-cli/node_modules/node-gyp',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '-Dnode_lib_file=/Users/runner/Library/Caches/node-gyp/18.18.2/<(target_arch)/node.lib',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '-Dmodule_root_dir=/Users/runner/work/amplify-cli/amplify-cli/node_modules/node-pty',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '-Dnode_engine=v8',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '--depth=.',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '--no-parallel',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '--generator-output',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   'build',
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args   '-Goutput_dir=.'
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp info spawn args ]
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR Traceback (most recent call last):
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR   File "/Users/runner/work/amplify-cli/amplify-cli/node_modules/node-gyp/gyp/gyp_main.py", line 42, in <module>
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR     import gyp  # noqa: E[40](https://github.com/aws-amplify/amplify-cli/actions/runs/6802605462/job/18496166607#step:3:43)2
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR     ^^^^^^^^^^
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR   File "/Users/runner/work/amplify-cli/amplify-cli/node_modules/node-gyp/gyp/pylib/gyp/__init__.py", line 9, in <module>
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR     import gyp.input
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR   File "/Users/runner/work/amplify-cli/amplify-cli/node_modules/node-gyp/gyp/pylib/gyp/input.py", line 19, in <module>
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR     from distutils.version import StrictVersion
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR ModuleNotFoundError: No module named 'distutils'
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp ERR! configure error 
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp ERR! stack Error: `gyp` failed with exit code: 1
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp ERR! stack     at ChildProcess.onCpExit (/Users/runner/work/amplify-cli/amplify-cli/node_modules/node-gyp/lib/configure.js:325:16)
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp ERR! stack     at ChildProcess.emit (node:events:517:28)
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:292:12)
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp ERR! System Darwin 21.6.0
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp ERR! command "/usr/local/Cellar/node@18/18.18.2/bin/node" "/Users/runner/work/amplify-cli/amplify-cli/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp ERR! cwd /Users/runner/work/amplify-cli/amplify-cli/node_modules/node-pty
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp ERR! node -v v18.18.2
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp ERR! node-gyp -v v9.3.1
  ➤ YN0000: │ node-pty@npm:0.11.0-beta19 [00181] STDERR gyp ERR! not ok 
  ➤ YN0009: │ node-pty@npm:0.11.0-beta19 [00181] couldn't be built successfully (exit code 1, logs can be found here: /private/var/folders/3s/vfzpb5r[51](https://github.com/aws-amplify/amplify-cli/actions/runs/6802605462/job/18496166607#step:3:54)gs6y328rmlgzm7c0000gn/T/xfs-e[63](https://github.com/aws-amplify/amplify-cli/actions/runs/6802605462/job/18496166607#step:3:66)ef17a/build.log)
```